### DISCUSSION TITLE
tcl interface to C++ API via Swig - user defined types

### DISCUSSION THREAD

**Discussion by Mercury-Fred** [swig/swig#3048] [Q&A] [answered]
Hi all,

I'm sure this question is extremely basic - but sorry I'm new to tcl and swig.  I am attempting to interface an existing c++ library to tcl via swig.  I've been able to build the wrapper and import/load into swig successfully.  I am attempting to use the API without writing any new APIs that might be customized to make interfacing with tcl easier.   So, for example, some of my apis take parameters of type int32_t, a simple signed 32 bit integer.  When I attempt to invoke these methods via tcl, I get a "TypeError in method ... argument 2 of type 'int32_t'.  I can't figure out how to force the type of my second parameter to be compatible with what tcl expects.  I believe I've been able to create an integer variable... but if I try to pass that to the method I still get the TypeError message.  Thus, tcl must be pretty strict with regard to matching the type of the tcl variable with that which the C++ API expects.

I've done plenty of googling but can't identify any resources for how user defined types in C++ can be mirrored/made compatible with tcl variables.  I see that most tcl variables are string by default, but can be forced to be other types, such as integer.. but don't see any way for tcl to create a type such as int32_t.

Thanks for any assistance!

**Comment by wsfulton** [comment, marked as answer]
The int32_t type is not available by default by C compilers, you have to
include the stdint.h header. Similarly for SWIG, it requires an include:

%include <stdint.i>

Then you'll find it'll work.

William
